pkg/types/validation/installconfig.go

```diff
@@ -596,13 +596,22 @@ func validateNetworkingClusterNetworkMTU(c *types.InstallConfig, fldPath *field.
 	return allErrs
 }
 
-func validateControlPlane(platform *types.Platform, pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
+func validateControlPlane(platform *types.Platform, pool *types.MachinePool, isSingleNodeOpenShift bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if pool.Name != types.MachinePoolControlPlaneRoleName {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), pool.Name, []string{types.MachinePoolControlPlaneRoleName}))
 	}
-	if pool.Replicas != nil && *pool.Replicas == 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
+	if pool.Replicas == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be provided"))
```

Comment:
In practice I don't think this can happen because of https://github.com/openshift/installer/blob/master/pkg/types/defaults/installconfig.go#L63
https://github.com/openshift/installer/blob/master/pkg/types/defaults/machinepools.go#L14-L16